plugins/inputs/logfile/fileconfig.go

```diff
@@ -203,3 +251,55 @@ func (config *FileConfig) isMultilineStart(logValue string) bool {
 	}
 	return config.MultiLineStartPatternP.MatchString(logValue)
 }
+
+func (config *FileConfig) shouldFilterLog(event logs.LogEvent) bool {
+	if config.Filters == nil || len(config.Filters) < 1 {
+		return false
+	}
+	config.sampleCount += 1
+	start := time.Now()
+	for _, filter := range config.Filters {
+		if filter.ExpressionP == nil {
+			filter.ExpressionP = regexp.MustCompile(filter.Expression)
+		}
+		matched := filter.ExpressionP.MatchString(event.Message())
+		switch filter.Type {
+		case includeType:
+			if matched {
+				elapsed := time.Since(start)
+				if config.sampleCount >= sampleThreshold {
+					log.Printf("D! [inputs.logfile.%s.%s] Matched log message with expression {%s} in %dns", config.LogGroupName, config.LogStreamName, filter.Expression, elapsed.Nanoseconds())
+					config.sampleCount = 0
+				}
+				return false
```

Comment:
If there are 5 LogFilters, all of which are INCLUDE type, then this code would check the first LogFilter and if it matches it returns false.
Is it allowable for a user to configure 2 or more INCLUDE type filters?
If so, then you cannot return before checking all filters.

Reply:
I'm not sure I follow. 

Yes, it is allowable for a user to configure more than one INCLUDE type filter. I believe that the easiest implementation for the log filter order of operations is to short circuit when we find a first match. What is the use case where we would match one include filter, and still need to check a subsequent include filter? By design, the filter configuration is an ordered list. The order of the list is used to determine the order in which the agent processes filters.

Reply:
I confused myself too.

Example config:
```
{
            "file_path": "/var/log/foo.log",
            "log_group_name": "bar",
            "log_stream_name": "baz",
            "filters": [
              {
                "type": "include",
                "expression": "(ERROR|error)"
              },
              {
                "type": "include",
                "expression": "(CRITICAL|critical)"
              }
            ]
          },
```

In this case I want to upload log lines that match filter_1 *OR*** filter_2.

At first I was thinking I would only want upload log lines that match filter_1 *AND*** filter_2.

So I am fine with short circuit `return false` on this line.

... Hopefully nobody else gets confused like I did :)

Reply:
Yeahhhhh I think that semantically I could have inverted the boolean so that it returns `true` when it's matched and we want to return early, but I don't think the structure would be changed much. It does what I want it to do; the return value is just kind of unintuitive. I'm open to flipping the condition and changing it from `shouldFilterLog` to something like `shouldPublishLog`, if you think that could help. Otherwise I think I'll keep it as is.

Reply:
Based on Ben's review, I think the suggestion is to go to what you were thinking originally where it's an AND condition. So, a lot of this conversation is moot. But I will make an effort to document the desired behavior.